src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/Program.cs

```diff
@@ -31,13 +32,20 @@ private void ProcessRunOptions(string[] args)
             //  -   # factories to pool
             //  -   # channels to pool
             //  -   etc
-            TestHelpers.SetHostAndProtocol(useHttp: false, hostName: "localhost", appName: "WcfService1");
+            var hostName = Environment.GetEnvironmentVariable("hostName");
+            if (String.IsNullOrEmpty(hostName))
+            {
+                hostName = "localhost";
+            }
+            Console.WriteLine("hostName: " + hostName);
+
+            TestHelpers.SetHostAndProtocol(useHttp: false, hostName: hostName, appName: "WcfService1");
         }
 
-        private void DoTheRun(int iterations)
+        private static void DoTheRun(int iterations)
         {
             Console.WriteLine("Start");
-            Task[] allTasks = new Task[23];
+            Task[] allTasks = new Task[13];
             for (int t = 0; t < allTasks.Length; t++)
```

Comment:
Nit: not required for merging, but magic numbers like '23' should become 'const' or something so we understand what they mean. It might also become a TestProperty we pass in from the outside if there is any value in that.